### ISSUE TITLE
Missing XCTest when included in framework podspec

### ISSUE THREAD

**Issue by chnewbry** [CocoaPods/CocoaPods#7727]
🌈

* [ ] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

1) Run 
``` pod lib create IQTestUtils ```

2) Add 'iOSSnapshotTestCase' as a dependency to my podspec
3) Add 'XCTest' as a framework to my podspec

changes to default podspec are:
```
s.dependency 'iOSSnapshotTestCase'
s.framework = 'XCTest'
```

4) cd into the example project and pod install
5) Run example project

## What did you expect to happen?

Project to successfully run

## What happened instead?

``` 
dyld: Library not loaded: @rpath/XCTest.framework/XCTest
  Referenced from: /Users/chnewbry/Library/Developer/CoreSimulator/Devices/AC364A95-0D42-4AA9-9A6B-A912849D8165/data/Containers/Bundle/Application/3909442E-FE59-4675-A3C3-2B9A6E3B8C2A/IQTestUtils_Example.app/Frameworks/FBSnapshotTestCase.framework/FBSnapshotTestCase
  Reason: image not found 
```

### Stack

```
   CocoaPods : 1.5.2
        Ruby : ruby 2.3.3p222 (2016-11-21 revision 56859) [universal.x86_64-darwin17]
    RubyGems : 2.5.2
        Host : Mac OS X 10.13.4 (17E199)
       Xcode : 9.3 (9E145)
         Git : git version 2.14.1
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 712149279d54d96f8488c856bbcb0db7bc3751b0
               mobiledatalabs - https://github.com/MobileDataLabs/MDLCocoapodSpecs.git @ 3845a21b7c5147b4efe34cc0ac21fe99b1da59d2
               visualstudio-cocoapods - ssh://msblox@msblox.visualstudio.com/DefaultCollection/_git/CocoaPods @ e3200b9af2b14ae8100c477f8418414edb1b9272
               visualstudio-cocoapods-1 - ssh://msblox@vs-ssh.visualstudio.com:22/DefaultCollection/_ssh/CocoaPods @ e7f20baff0ad965eaeea34ef616ad1deb4d97850
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
use_frameworks!

target 'IQTestUtils_Example' do
  pod 'IQTestUtils', :path => '../'

  target 'IQTestUtils_Tests' do
    inherit! :search_paths

    
  end
end
```

## Project that demonstrates the issue

https://github.com/chnewbry/IQTestUtils

## Potentially Related Issue
https://github.com/CocoaPods/CocoaPods/issues/5293

My understanding is that iOSSnapshotTestCase has a dependency on XCTest as demonstrated in their Podspec with: 

```s.framework = 'XCTest'```

Based on the issue linked above I think I'd need to explicitly mark IQTestUtils as depending on the XCTest framework as well. 

Please let me know if this isn't the right forum for this issue. I'm unsure if it's a bug in CocoaPods or (probably more likely) programmer error on my part. If so I can move this question over to SO. 

🙏 Thanks for your time!

**Comment by segiddins** [comment]
Thanks for the issue! I think this might be fixed on master, but no promises. I’ll try to check when I get a chance

**Comment by chnewbry** [comment]
@segiddins I gave cloning the master branch of cocoapods a shot based on the documentation found here:

https://guides.cocoapods.org/using/unreleased-features

I followed the "Setup a local dev copy of Cocoapods" instructions. 

I did this two weeks ago when you messaged me about it. Wondering if you have any other suggestions for me to look into? I'm happy to poke around I can't think of other things to try at the moment.

**Comment by dnkoutso** [comment]
@chnewbry easiest way is to use Bundler and pin to a specific SHA https://bundler.io/v1.12/git.html

Please close this issue if it has been fixed for you on latest master.

**Comment by chnewbry** [comment]
@dnkoutso thanks for the comment 🙏 

Tested on latest master getting the same error as outlined above.

**Comment by nickromano** [comment]
If you're creating a pod for testing you'll need to move the pod reference to the test target.

Instead of this
```swift
use_frameworks!

target 'IQTestUtils_Example' do
  pod 'IQTestUtils', :path => '../'

  target 'IQTestUtils_Tests' do
    inherit! :search_paths

    
  end
end
```

You'll want this
```swift
use_frameworks!

target 'IQTestUtils_Example' do

  target 'IQTestUtils_Tests' do
    inherit! :search_paths

    pod 'IQTestUtils', :path => '../'
  end
end
```

**Comment by dnkoutso** [comment]
Seems fixed for 1.6.0 (not released) yet.

![screen_shot_2018-05-25_at_1_30_23_pm](https://user-images.githubusercontent.com/310370/40565020-d015b104-601f-11e8-9153-7a2fb6d8d201.png)

Recommend using https://bundler.io/v1.12/git.html to point to this SHA
